internal/controller/apiextensions/definition/indexes.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2023 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package definition
+
+import (
+	"context"
+	"fmt"
+
+	kunstructured "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	runtimeevent "sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	compositeresource "github.com/crossplane/crossplane-runtime/pkg/resource/unstructured/composite"
+)
+
+const (
+	// compositeResourceRefGVKsIndex is an index of all GroupKinds that
+	// are in use by a Composition. It indexes from spec.resourceRefs, not
+	// from spec.resources. Hence, it will also work with functions.
+	compositeResourceRefGVKsIndex = "compositeResourceRefsGVKs"
+	// compositeResourcesRefsIndex is an index of resourceRefs that are owned
+	// by a composite.
+	compositeResourcesRefsIndex = "compositeResourcesRefs"
+)
+
+// IndexCompositeResourceRefGVKs assumes the passed object is a composite. It
+// returns gvk keys for every resource referenced in the composite.
+func IndexCompositeResourceRefGVKs(o client.Object) []string {
```

Comment:
Is this and `IndexCompositeResourcesRefs` satisfying some interface? If so, is it worth testing that it does (mostly for documentation purposes)?

Reply:
Not an interface, just a function type.

Reply:
Added a type assertion.